solidity/contracts/Coordinator.sol

```diff
@@ -184,4 +187,51 @@ contract Coordinator {
     require(_amount >= serviceAgreements[_sAId].payment, "Below agreed payment");
     _;
   }
+
+  modifier hasInternalId(uint256 _internalId) {
+    require(callbacks[_internalId].addr != address(0), "Must have a valid internalId");
+    _;
+  }
+
+  function fulfillData(
+    uint256 _internalId,
+    bytes32 _data
+  )
+    public
+    onlyOwner
```

Comment:
Instead of `onlyOwner` here we could remove it entirely and add a new story for "only accept responses from valid SA oracles".

Reply:
Done. Story here: https://www.pivotaltracker.com/story/show/161866471